File: python/007_projecteuler.py
def is_prime(num):
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

def find_prime(n):
    count = 0
    num = 2

    while count < n:
        if is_prime(num):
            count += 1
            if count == n:
                return num
            
        num += 1

result = find_prime(10001)
print(result)

File: C/007_projecteuler.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>  // Includes sqrt

bool isprime(int num) {
    if (num <= 1) return false;
    if (num == 2) return true; // Only even prime number :D

    for (int i = 2; i <= sqrt(num); i++) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

int findprime(int n) { // Finds the nth prime
    int count = 0; // Primes found
    int num = 1;

    while (count < n) {
        num += 1;
        if (isprime(num)) {
            count += 1;
        }
    }
    return num;
}

int main() {
    int result = findprime(1001);
    printf("%d\n", result);
    return 0;
}